### ISSUE TITLE
Deployment fails when namespace has istio-injection=enabled

### ISSUE THREAD

**Issue by jambajaar** [rancher/rancher#29700]
I'm trying out Istio on our dev cluster with Rancher 2.5 and I can't deploy anything to a namespace when sidecar injection is enabled. I get the following error:

Pods "alpine-688478fb48-" is forbidden: unable to validate against any pod security policy: \[spec.securityContext.fsGroup: Invalid value: \[\]int64{1337}: group 1337 must be in the ranges: \[{0 1024}\]\]

I assume the sidecar proxy is using fsGroup 1337 because the alpine image I'm trying to deploy for testing doesn't. Why does the proxy need to set fsGroup? Can I change this in the Istio config? 

gz#12851

**Comment by oxr463** [comment]
This appears to be an instance of this issue in upstream Istio, (See: https://github.com/istio/istio/issues/26882), which depends on https://github.com/kubernetes/kubernetes/issues/57923.

**Comment by BryanHunt** [comment]
Additionally, Istio is using UID 1337 which overlaps with our NIS UIDs and creates a security hole when mounting NFS volumes.

**Comment by brendarearden** [comment]
@jambajaar what version of kubernetes is the dev cluster running on?

**Comment by oxr463** [comment]
Update from Istio upstream issue, (See: https://github.com/istio/istio/issues/26882#issuecomment-736115147):

> This is fixed in Kubernetes 1.19 by #27367. It will automatically stop setting this field if you use 1.19. I don't think its possible to fix on older versions.

**Comment by brendarearden** [comment]
Starting in Istio 1.8, https://github.com/istio/istio/pull/27367 a flag as been added to ensure the fsGroup does not get set. We are not currently supporting 1.8. This is does not get set by default in kubernets 1.19.x

The solution for using Istio 1.7.x without the fsGroupd errors is to use kubernetes 1.19.x. This can be tested by running Istio 1.7.x on a Kubernetes < 1.19.x, observe the error occur when enabling istio-injection and then attempting the same steps in a 1.19.x environment. 

The reason this did not occur previously and is occuring now is due to the way we are installing istio starting in 1.7.x. While the install occurs with a helm chart in the ui, it is using the istioctl under the hood.

**Comment by cbron** [comment]
Moving to test to ensure this can be fixed by upgrading to 1.19. There aren't any code changes on our side. We currently don't support 1.8 but when we do that path should also fix.

**Comment by bmdepesa** [comment]
rancher/rancher:v2.5-head 8d759be

Reopening this as upgrading the cluster did not resolve the issue. 

I reproduced with the following steps: 
* Create a 1.17.x cluster w/ PSP enabled
* Modify the PSP to reflect the user's: `FS Group Policy` set to `MustRunAs` with a range of `1..1024`
* Install Istio & enable istio-injection on a namespace
* Deploy a workload to the namespace and see the pod forbidden to run:
```
unable to validate against any pod security policy: [spec.securityContext.fsGroup: Invalid value: []int64{1337}: group 1337 must be in the ranges: [{0 1024}]]
```

* I then upgraded the cluster to `1.19.4`
* Attempt to deploy a workload with the same result. 

Looking at the referenced Istio issues, it seems like the fix will only be in Istio 1.8, and while the fix will require a flag set in k8s < 1.19, it will automatically apply if the cluster is 1.19+ (but still requires Istio 1.8) .

**Comment by bmdepesa** [comment]
rancher/rancher:v2.5.6-rc5
`rancher-istio:v1.8.3-rc01`
k8s 1.20 cluster

* Create a cluster w/ PSP enabled
* Deploy Istio
* Copy Rancher's `unrestricted` PSP and modify `FS Group Policy set to MustRunAs with a range of 1..1024`
* Enable istio-injection on a namespace where the PSP with `FS Group Policy` is being enforced
* Deploy the Istio demo bookservice
* The app can be deployed

Other notes:
* The fsGroup is set to `1` currently, whereas it was previously `1337` due to the upstream Istio changes
* `values.pilot.env.ENABLE_LEGACY_FSGROUP_INJECTION=false` (from upstream https://github.com/istio/istio/pull/27367/commits/7ffaa70a588f0e2114fbe7475b5264af4f0ec0ad)  is set correctly in the Istio deployment
